packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -43,380 +46,399 @@ import {
   GRID_ROW_REACT_KEY_PROPERTY,
   removeNoOpOperations,
 } from './utils/DataGridUtils'
-import { StartGridSession } from './StartGridSession'
+import { StartGridSession, StartGridSessionHandle } from './StartGridSession'
 import { useDataGridWebSocket } from './useDataGridWebsocket'
 import { useGridUndoRedo } from './hooks/useGridUndoRedo'
 import { applyModelChange, ModelChange } from './utils/applyModelChange'
 import { mapOperationsToModelChanges } from './utils/mapOperationsToModelChanges'
 import { getCellClassName } from './utils/getCellClassName'
 
 export type SynapseGridProps = {
-  query: string
   showDebugInfo?: boolean
 }
 
-const SynapseGrid = forwardRef<
-  { initializeGrid: () => void },
-  SynapseGridProps
->(({ query, showDebugInfo = false }, ref) => {
-  const [session, setSession] = useState<GridSession | null>(null)
-  const [replicaId, setReplicaId] = useState<number | null>(null)
-  const [presignedUrl, setPresignedUrl] = useState<string>('')
-
-  const startGridSessionRef = useRef<{
-    handleStartSession: (input: string) => void
-  } | null>(null)
-
-  useImperativeHandle(
-    ref,
-    () => ({
-      initializeGrid: () => {
-        if (startGridSessionRef.current) {
-          startGridSessionRef.current.handleStartSession(query)
-        }
+export type SynapseGridHandle = {
```

Comment:
Suggest hiding whitespace changes -- the code formatter reduced indentation in this file, creating a large diff.